core/include/mmcore/view/BaseView.h

```diff
@@ -0,0 +1,410 @@
+                        if (this->_autoSaveCamSettingsSlot.template Param<param::BoolParam>()->Value()) {
+                            this->onStoreCamera(this->_storeCameraSettingsSlot); // manually trigger the storing
+                        }
+                    } else {
+                        if (this->_savedCameras[index].second) {
+                            this->_camera = this->_savedCameras[index].first; // override current camera
+                        }
+                    }
+                }
+
+                this->_camera_controller.OnKey(key, action, mods);
+            }
+
+            return false;
+        }
+
+        template<typename VIEWCALL_TYPE, typename CAM_CONTROLLER_TYPE>
+        inline bool BaseView<VIEWCALL_TYPE, CAM_CONTROLLER_TYPE>::OnChar(
+            unsigned int codePoint) {
+            auto* cr = this->_rhsRenderSlot.template CallAs<AbstractCallRender>();
+            if (cr == NULL)
+                return false;
+
+            InputEvent evt;
+            evt.tag = InputEvent::Tag::Char;
+            evt.charData.codePoint = codePoint;
+            cr->SetInputEvent(evt);
+            if (!(*cr)(AbstractCallRender::FnOnChar))
+                return false;
+
+            if (this->_cameraIsMutable) {
+                this->_camera_controller.OnChar(codePoint);
+            }
+
+            return true;
+        }
+
+        template<typename VIEWCALL_TYPE, typename CAM_CONTROLLER_TYPE>
+        inline bool
+        BaseView<VIEWCALL_TYPE, CAM_CONTROLLER_TYPE>::OnMouseButton(
+            view::MouseButton button, view::MouseButtonAction action, view::Modifiers mods) {
+            if (!this->_camera_controller.isOverriding() && !this->_camera_controller.isActive()) {
+                auto* cr = this->_rhsRenderSlot.template CallAs<AbstractCallRender>();
+                if (cr != nullptr) {
+                    view::InputEvent evt;
+                    evt.tag = view::InputEvent::Tag::MouseButton;
+                    evt.mouseButtonData.button = button;
+                    evt.mouseButtonData.action = action;
+                    evt.mouseButtonData.mods = mods;
+                    cr->SetInputEvent(evt);
+                    if ((*cr)(AbstractCallRender::FnOnMouseButton))
+                        return true;
+                }
+            }
+
+            if (this->_cameraIsMutable) {
+                // get window resolution to help computing mouse coordinates
+                int wndWidth;
+                int wndHeight;
+                auto projType = this->_camera.template get<Camera::ProjectionType>();
+                if (projType == Camera::ProjectionType::PERSPECTIVE) {
+                    auto tile_end = this->_camera.template get<Camera::PerspectiveParameters>().image_plane_tile.tile_end;
+                    auto tile_start = this->_camera.template get<Camera::PerspectiveParameters>().image_plane_tile.tile_start;
```

Comment:
The `wndWidth` and `wndHeight` computation looks somewhat copy-pasted here and in `OnMouseMove()`. Can we minimize room for error by re-using lambdas or something like that?